app/components/app_activity_log_component.rb

```diff
@@ -0,0 +1,26 @@
+class AppActivityLogComponent < ViewComponent::Base
+  def initialize(patient_session)
+    super
+
+    @patient_session = patient_session
+  end
+
+  def events_by_day
+    consent_events.group_by { _1[:time].to_fs(:nhsuk_date) }
+  end
+
+  def consent_events
+    @patient_session
+      .patient
+      .consents
+      .recorded
+      .order(recorded_at: :desc)
+      .map do |consent|
+        {
+          title:
+            "Consent #{consent.response} by #{consent.parent_name} (#{consent.who_responded})",
```

Comment:
Just noticing the who_responded can be blank in our test data, is that realistic?

![image](https://github.com/nhsuk/manage-vaccinations-in-schools/assets/15608/5758fff4-23af-4306-b97b-2b835d416d77)

Reply:
This shouldn't ever be blank. @misaka can you figure out what the test data is missing, so we might fix it in the generator? e.g. we not populating `consents#parent_relationship_other` if `consents#parent_relationship` is equal to `other`...

Also "Consent #{consent.response} by #{consent.parent_name} (#{consent.who_responded})" will show the wrong thing for self-consent, I think.

Reply:
Yeah, it's a problem with the data. I'll look into this 👍

Reply:
Thanks – there's also the 2nd part which I think is still an issue in the code:

> Also "Consent #{consent.response} by #{consent.parent_name} (#{consent.who_responded})" will show the wrong thing for self-consent, I think.

Reply:
Ya, I realised this as well when I tried a different approach to fixing the factories, I tried adding validations that worked outside of the wizard steps. That caught out that self-consent didn't set `parent_relationship`.

I think we should have validations that work outside of the wizard steps to catch these issues.